diff --git c/mypy.bzl i/mypy.bzl
index e5243f1..32f5fb0 100644
--- c/mypy.bzl
+++ i/mypy.bzl
@@ -99,6 +99,12 @@ def _extract_imports(imports, label):
             mypypath_parts.append("{}/{}".format(label.package, import_))
     return mypypath_parts
 
+def _sanitize_for_dirname(name):
+    # Sanitizes a string (with input expected to be a Bazel target) to be safe
+    # to use in a directory name.
+    # Replace problematic characters with underscores
+    return name.replace("/", "_").replace(":", "_").replace("@", "_")
+
 def _mypy_rule_impl(ctx, is_aspect = False):
     base_rule = ctx
     if is_aspect:
@@ -164,6 +170,8 @@ def _mypy_rule_impl(ctx, is_aspect = False):
         substitutions = {
             "{MYPY_EXE}": ctx.executable._mypy_cli.path.replace("external", "..", 1),
             "{MYPY_ROOT}": ctx.executable._mypy_cli.root.path.replace("external", "..", 1),
+            # Cache map triple computation kept to limit divergence with upstream,
+            # but they are not used anymore.
             "{CACHE_MAP_TRIPLES}": " ".join(_sources_to_cache_map_triples(src_files, is_aspect)),
             "{PACKAGE_ROOTS}": " ".join([
                 "--package-root " + shell.quote(path or ".")
@@ -172,12 +180,17 @@ def _mypy_rule_impl(ctx, is_aspect = False):
             "{SRCS}": " ".join([
                 shell.quote(f.path) if is_aspect else shell.quote(f.short_path)
                 for f in src_files
+                # Filter out generated pb2 proto files from source file set to test explicitly via mypy,
+                # as listing these explicitly will make mypy ignore .pyi files in the same package
+                # [these are still included and tested as transitive deps]
+                if not (f.path.endswith("_pb2.pyi") or f.path.endswith("_pb2.py") or f.path.endswith("_pb2_grpc.py"))
             ]),
             "{VERBOSE_OPT}": "--verbose" if DEBUG else "",
             "{VERBOSE_BASH}": "set -x" if DEBUG else "",
             "{OUTPUT}": out.path if out else "",
             "{MYPYPATH_PATH}": mypypath if mypypath else "",
             "{MYPY_INI_PATH}": mypy_config_file.path.replace("external", "..", 1),
+            "{CACHE_DIR}": "${MYPY_CACHE_DIR:-.}/" + _sanitize_for_dirname(ctx.label.package + "/" + ctx.label.name),
         },
         is_executable = True,
     )
diff --git c/mypy/BUILD i/mypy/BUILD
index f5cb9c4..f7b5937 100644
--- c/mypy/BUILD
+++ i/mypy/BUILD
@@ -1,3 +1,7 @@
+exports_files([
+    "main.py",
+])
+
 py_binary(
     name = "mypy",
     srcs = ["main.py"],
diff --git c/templates/mypy.sh.tpl i/templates/mypy.sh.tpl
index 0b01b3b..c348978 100644
--- c/templates/mypy.sh.tpl
+++ i/templates/mypy.sh.tpl
@@ -27,7 +27,14 @@ main() {
   fi
 
   set +o errexit
-  output=$($mypy {VERBOSE_OPT} --bazel {PACKAGE_ROOTS} --config-file {MYPY_INI_PATH} --cache-map {CACHE_MAP_TRIPLES} -- {SRCS} 2>&1)
+  # --bazel and --cache-map parameters present upstream have been removed since
+  # the approach they implemented wasn't compatible with the Bazel sandbox as
+  # described here:
+  # https://github.com/bazel-contrib/bazel-mypy-integration/issues/48
+  #
+  # Instead we maintain a separate, standard MyPy cache for each test, whose
+  # location is passed in through --cache-dir.
+  output=$($mypy {VERBOSE_OPT} --cache-dir {CACHE_DIR} {PACKAGE_ROOTS} --config-file {MYPY_INI_PATH} -- {SRCS} 2>&1)
   status=$?
   set -o errexit
 
