--- a/triton/backends/nvidia/driver.py
+++ b/triton/backends/nvidia/driver.py
@@ -31,6 +31,12 @@ def libcuda_dirs():
     env_ld_library_path = os.getenv("LD_LIBRARY_PATH")
     if env_ld_library_path and not dirs:
         dirs = [dir for dir in env_ld_library_path.split(":") if os.path.exists(os.path.join(dir, "libcuda.so.1"))]
+    env_ld_preload = os.getenv("LD_PRELOAD")
+    if env_ld_preload and not dirs:
+        preload_dirs = [Path(lib_path).parent for lib_path in env_ld_preload.split(":") if Path(lib_path).name.startswith("libcuda.so")]
+        if len(preload_dirs) > 0:
+            dirs = preload_dirs
+
     msg = 'libcuda.so cannot found!\n'
     if locs:
         msg += 'Possible files are located at %s.' % str(locs)

--- a/triton-3.3.0.dist-info/RECORD
+++ b/triton-3.3.0.dist-info/RECORD
@@ -114,7 +114,7 @@
 triton/backends/nvidia/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 triton/backends/nvidia/compiler.py,sha256=eJld1AxOVJZPYs3G1H5m9tnfi1Wxcq1-uZI18nmZyEA,19427
 triton/backends/nvidia/driver.c,sha256=a1EQkPp1bnC6AbMyAOYUmeIWzU-z_w52N_yJ1JhMJR4,17277
-triton/backends/nvidia/driver.py,sha256=98Juj_2vjNJJHeBCF9MoJ7Gk-Fn3-T6lbiZZDJ9eBxo,19911
+triton/backends/nvidia/driver.py,sha256=JV7FEJhre_djTgs7--eQDTa0__InDtcu8eb_7SNjdyI,20196
 triton/backends/nvidia/bin/cuobjdump,sha256=rR0PBpn0ZgNBbrWOe5_fmik-lfkxLJ33_W_aVqbDDUE,569008
 triton/backends/nvidia/bin/nvdisasm,sha256=EIC8kJowIHYf9MeKm3vZtJN725wJeHCcvlI4HeYSofI,5894744
 triton/backends/nvidia/bin/ptxas,sha256=UDKqXtwKmWD7QJ3xwURFw4XHhnjGNC66W8Q20ePTUgs,30309984